### PR TITLE
Implement basic UI

### PR DESCRIPTION
**Pull request by rehandalal**
Once #2 has been merged and this has been rebased the UI will be usable.

f?

### CODE REVIEW

File: morgoth/base/static/js/actions/addons.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import axios from 'axios';
```
Comment (mythmon) [inline morgoth/base/static/js/actions/addons.js]: axios looks really similar to the almost-standard fetch API. Any reason not to just use fetch?

File: morgoth/base/static/js/components/Page.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  palette: {
+    primary1Color: blueGrey500,
+    primary2Color: blueGrey700,
+    primary3Color: blueGrey400,
+    accent1Color: orange800,
+  },
+});
+
+const Page = props => {
+  const { routes, children } = props;
```
Comment (mythmon) [inline morgoth/base/static/js/components/Page.jsx]: I think this pattern is better as `const Page = ({ routes, children }) => {`.

I'm also not a huge fan of `const Foo = () => {}` vs `function Foo() {}`, but I have no good reasons for that.

Comment (rehandalal) [inline morgoth/base/static/js/components/Page.jsx]: agreed about `const Page = ({ routes, children }) => {`

and I don't have a preference between the latter but I'm happy to switch. I do think we should keep it consistent.

File: morgoth/base/static/js/actions/addons.js (hunk 1 above)
Comment (rehandalal) [inline morgoth/base/static/js/actions/addons.js]: I guess I could switch to just using fetch. I suspect the limitations of fetch won't be an issue for us.

**Comment by rehandalal** [comment]
@brittanystoroz your input is welcome too!

**Comment by rehandalal** [comment]
i think this is as far as i want to take this PR. in the effort of keeping it from becoming crazy large.

r?

File: morgoth/base/static/js/containers/AddonsListContainer.js [hunk 3]
```diff
@@ -13,13 +13,20 @@ function mapStateToProps(state) {
...
-          dispatch(fetchAddonsFailure(response.payload));
-        } else {
-          dispatch(fetchAddonsSuccess(response.payload));
-        }
-      });
+      dispatch(fetchAddons())
+        .then(response => {
+          response.payload.json().then(jsonData => {
+            const data = jsonData === '' ? {} : jsonData;
+            if (response.payload.ok) {
```
Comment (mythmon) [inline morgoth/base/static/js/containers/AddonsListContainer.js]: I'd usually avoid a level of indentation and write this like:

```javascript
dispatch(fetchAddons())
.then(response => [response, response.json()])
.then(([response, jsonData]) => {
  const data = jsonData || '';
  ...
});
```

Though that isn't a lot better. This is one problem with the fetch API, unfortunately.

File: morgoth/base/static/js/reducers/addonsReducer.js [hunk 4]
```diff
@@ -12,8 +12,6 @@ const INITIAL_STATE = {
 };
 
 export default (state = INITIAL_STATE, action) => {
-  let error;
```
Comment (mythmon) [inline morgoth/base/static/js/reducers/addonsReducer.js]: By the way, to deal with variables in reducers, I found this pattern to work well:

```javascript
export default (state = INITIAL_STATE, action) => {
  switch (action.type):
    case FETCH_ADDONS_FAILURE: {
      let error = action.payload;
      return {
        ..state,
        addonsList: {
          error,
          loading: false,
        };
    }
}
```

File: morgoth/base/static/js/utils/apiFetch.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+      throw new Error('Only pass one of `settings.data` and `settings.body`.');
+    }
+
+    const method = (settings.method || 'get').toLowerCase();
+
+    if (method !== 'get' && method !== 'head') {
+      settings.body = JSON.stringify(settings.data);
+    }
+
+    delete settings.data;
```
Comment (mythmon) [inline morgoth/base/static/js/utils/apiFetch.js]: Probably isn't a good idea to just throw away `settings.data` silently if method is "get" or "head".

File: morgoth/base/static/js/utils/apiFetch.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    throw new Error('A valid `success` function must be provided.');
+  }
+
+  // Extract the `handleError` function from settings.
+  if (typeof settings.error === 'function') {
+    handleError = settings.error;
+    delete settings.error;
+  } else {
+    throw new Error('A valid `error` function must be provided.');
+  }
```
Comment (mythmon) [inline morgoth/base/static/js/utils/apiFetch.js]: Have you considered abusing proptypes for this sort of validation? (Though it isn't actually very good at it...). Either way, it seems like this is something that could be improved.

File: morgoth/base/static/js/utils/apiFetch.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+  return fetch(`${API_ROOT}${url}`, settings)
+    .then(response => Promise.all([response.ok, response.json()]))
+    .then(values => {
+      const ok = values[0];
+      const data = values[1];
```
Comment (mythmon) [inline morgoth/base/static/js/utils/apiFetch.js]: You can put this in the parameter list too

```javascript
.then(([ok, data]) => {
```

should work just fine

**Comment by mythmon** [comment]
This looks fine to me. A couple nits, but otherwise r+

File: morgoth/base/static/js/utils/apiFetch.js (hunk 6 above)
Comment (rehandalal) [inline morgoth/base/static/js/utils/apiFetch.js]: Yeah I'd like to improve this. I'll add a TODO to look at improving this later.

**Pull request merged**